ui/segmented-control.mod/segmented-control.js

```diff
@@ -73,13 +75,32 @@ const SegmentedControl = (exports.SegmentedControl = class SegmentedControl exte
     enterDocument() {
         this._cancelHandleOptionsChange = this.addRangeAtPathChangeListener("_options", this, "handleOptionsChange");
         this.addPathChangeListener("_selectedOption", this, "handleSelectionChange");
+        this.element.addEventListener("change", this, {
+               size: {
+                     box: "border-box"
+              }
+        });
     }
-
     exitDocument() {
         this.removePathChangeListener("_selectedOption", this);
         this._cancelHandleOptionsChange?.();
     }
 
+    handleChange() {
+        this._isResizing = true;
+
+        if (this._resizeTimer) {
+            clearTimeout(this._resizeTimer);
+        }
+
+        this.needsDraw = true;
+
+        this._resizeTimer = setTimeout(() => {
```

Comment:
The use of a setTimeout() here really doesn't feel right, please discuss this with the team to find an alternative that does not involve the use of a timeout